File: Python/Garage.py
# import class car dari file Car
from Car import Car

# deklarasi kelas

class Garage:

    # atribut private
    __nama_garasi = ""
    __luas_garasi = ""
    # composite kelas Car dan motorcycle
    __car = Car()
    __motor = []  # array of object motorcycle

   # konstruktor
    def __init__(self, nama_garasi="", luas_garasi="", car=Car(), motor=[]):
        self.__nama_garasi = nama_garasi
        self.__luas_garasi = luas_garasi
        self._car = car
        self.__motor = motor

    # Getter & Setter

    def get_nama_garasi(self):
        return self.__nama_garasi

    def set_nama_garasi(self, nama_garasi):
        self.__nama_garasi = nama_garasi

    def get_luas_garasi(self):
        return self.__luas_garasi

    def set_luas_garasi(self, luas_garasi):
        self.__luas_garasi = luas_garasi

    def get_car(self):
        return self.__car

    def set_car(self, car):
        self.__car = car

    def get_motor(self):
        return self.__motor

    def set_motor(self, motor):
        self.__motor = motor

File: CPP/Garage.cpp
// import file header
#include "header.hh"

// konstruktor tanpa parameter
Garage::Garage() : Vehicle()
{
    // set isi atribut secara default
    this->nama_garasi = "";
    this->luas_garasi = "";
}
// konstruktor dengan parameter
Garage::Garage(string plat, string merk, string tahun_produksi, string warna, string nama_garasi, string luas_garasi, Motorcycle motorcycle, Car crr) : Vehicle(plat, merk, tahun_produksi, warna)
{
    // set isi atribut berdasarkan isi parameter
    this->nama_garasi = nama_garasi;
    this->luas_garasi = luas_garasi;
    this->motorcycle = motorcycle;
    this->crr.push_back(crr);
}

/* Getter dan Setter*/

// Get nama_garasi
string Garage::get_nama_garasi()
{
    return this->nama_garasi;
}

// Set nama_garasi
void Garage::set_nama_garasi(string nama_garasi)
{
    this->nama_garasi = nama_garasi;
}

// Get luas_garasi
string Garage::get_luas_garasi()
{
    return this->luas_garasi;
}

// Set luas_garasi
void Garage::set_luas_garasi(string luas_garasi)
{
    this->luas_garasi = luas_garasi;
}

// Get motorcycle
Motorcycle Garage::get_motorcycle()
{
    return this->motorcycle;
}

// Set motorcycle
void Garage::set_motorcycle(Motorcycle motorcycle)
{
    this->motorcycle = motorcycle;
}
// Get Car
list<Car> Garage::get_crr()
{
    return this->crr;
}

// Set Car
void Garage::set_crr(list<Car> crr)
{
    this->crr = crr;
}
// Add list objek Car
void Garage::add_crr(Car crr)
{
    this->crr.push_back(crr);
}


/* desktruktor */

// default
Garage::~Garage() // desktruktor
{
}
